### DISCUSSION TITLE
Mania mods: reduce redundancy between Fade In, Hidden, and Cover

### DISCUSSION THREAD

**Discussion by trustytrojan** [ppy/osu#27377] [Ideas] [answered]
Noticed in the past release Cover was added, the customization of FI and HD are gone, their stable functionality is back, and now only Cover has a constant coverage. My suggestion is to reduce the redundancy between these three mods by having one called "Hidden" and allowing the following customization options:
- Whether to slowly increase coverage on combo
- Coverage direction (already in Cover)
- Starting coverage (already in Cover but named "Coverage")

I think simplifying mods down into one with more customization is a key feature of Lazer so I hope to see that emphasized.

**Comment by bdach** [comment]
Merging FI and HD has implications for web (they have separate leaderboards, which would now be impossible, because there's no method to display a leaderboard based on mod settings, and I'm not sure if there will ever be). See also: https://github.com/ppy/osu/pull/25796#issuecomment-1859829241.

**Reply by trustytrojan** [reply]
I see, the current leaderboard system is a huge obstacle for something like this. Maybe only lazer scores (without CL) can have their FI/HD merged? Past stable scores can be left alone, and when a new stable score is submitted with either FI or HD the server will convert it to "HD + along scroll + increase cover on combo" or "HD + against scroll + increase cover on combo", respectively. This should be a lot less work than retroactively merging every single score on the stable leaderboards.

**Reply by bdach** [reply, marked as answer]
I'll say this: The operation is doable. Whether we're going to do it - dunno. It sure is going to require buy-in from more people than just me.
